src/xrGame/console_commands.cpp

```diff
@@ -1080,11 +1080,17 @@ class CCC_DebugFonts : public IConsole_Command
 {
 public:
     CCC_DebugFonts(LPCSTR N) : IConsole_Command(N) { bEmptyArgsHandled = true; }
+    ~CCC_DebugFonts() { xr_free(ui); }
     virtual void Execute(LPCSTR args)
     {
-        // BUG: leak
-        (xr_new<CUIDebugFonts>())->ShowDialog(true);
+        if (!ui)
+            ui = xr_new<CUIDebugFonts>();
```

Comment:
With this solution, there's an issue: CUIDebugFonts calls xr_delete on itself when you press Escape, so CCC_DebugFonts class will store an invalid pointer after the memory deletion.

Reply:
Is it possible to draw more than one instance of one font type? I'd like to draw letterica18 in two places of screen for example.

Reply:
I've tried to test this but using `debug_fonts` multiple times seems to work without crashing.
Also looking at the code for `HideDialog` I don't see a deallocation but I may be missing something here.

Reply:
Ohhh, if it was calling xr_delete on itself, then there would be no memory leak, but there was no xr_delete call.
I got mixed up a bit))